Client/src/Components/DashboardMenu/index.jsx

```diff
@@ -16,13 +18,56 @@ import SettingsIcon from "../../assets/Images/Icon-setting-gray.png";
  */
 
 function DashboardMenu() {
+  const [activeButton, setActiveButton] = useState(null);
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  useEffect(() => {
+    setActiveButton(location.pathname);
+  }, [location]);
+
+  const handleClick = (title, path) => {
+    setActiveButton(title);
+    navigate(path);
+  };
+
   return (
     <div className="dashboard-menu-container">
-      {DashboardMenuButton(Monitors, "Monitors", "/monitors")}
-      {DashboardMenuButton(Incidents, "Incidents", "/incidents")}
-      {DashboardMenuButton(SensorsIcon, "Status Pages", "/status")}
-      {DashboardMenuButton(AllInclusiveIcon, "Integrations", "/integrations")}
-      {DashboardMenuButton(SettingsIcon, "Settings", "/settings")}
+      {DashboardMenuButton(
+        Monitors,
+        "Monitors",
+        "/monitors",
+        activeButton === "/monitors",
+        () => handleClick("Monitors", "/monitors")
+      )}
+      {DashboardMenuButton(
+        Incidents,
+        "Incidents",
+        "/incidents",
+        activeButton === "/incidents",
+        () => handleClick("Incidents", "/incidents")
+      )}
+      {DashboardMenuButton(
+        SensorsIcon,
+        "Status Pages",
+        "/status",
+        activeButton === "/status",
+        () => handleClick("Status Pages", "/status")
+      )}
+      {DashboardMenuButton(
+        AllInclusiveIcon,
+        "Integrations",
+        "/integrations",
+        activeButton === "/integrations",
+        () => handleClick("Integrations", "/integrations")
```

Comment:
Rather than passing the path and the title, what about just passing the path and having a map for the title:

```
const pathMap = {
    "/integrations" : "Integrations",
    "/settings": "Settings",
    ...
}
``` 

You can then lookup the titles easily in the `handleClick` function:

```
const title = pathMap[path]
```

Then if we ever want to change the titles we just have to change the lookup map rather than going through each button.

What do you think @MuhammadKhalilzadeh ?

Reply:
That's a great idea, where do you want be to place the routing map?

Reply:
I'll probably add it in the same file or maybe create a new file for it in `utils` folder

Reply:
I think it's fine in the same file as it should only be used here I believe